src/main/java/org/cloudburstmc/server/Server.java

```diff
@@ -1717,41 +1718,43 @@ private void loadLevels() throws IOException {
         if (worldConfigs.isEmpty()) {
             throw new IllegalStateException("No worlds configured! Add a world to cloudburst.yml and try again!");
         }
-        List<CompletableFuture<Level>> levelFutures = new ArrayList<>(worldConfigs.size());
+        List<CompletableFuture<Level>> worldFutures = new ArrayList<>(worldConfigs.size());
 
         for (String name : worldConfigs.keySet()) {
             final ServerConfig.World config = worldConfigs.get(name);
-            //fallback to level name if no seed is set
             Object seedObj = config.getSeed();
             long seed;
+            if(seedObj == null) { //Auto generate seed if one was not specified
+                seedObj = ThreadLocalRandom.current().nextLong();
```

Comment:
i haven't actually looked at this code in-depth for a while, but won't this be run every time the server starts, thus causing the seed to be different between restarts? the seed isn't written to config anywhere, and i'm pretty sure the seed stored in `level.dat` isn't used...

Reply:
Whoops, looks like I forgot about that. I would go for level.dat personally, but does Cloudburst prefer config instead?